packages/graphiql/src/components/common/Resizer/ResizeHandler.tsx

```diff
@@ -0,0 +1,84 @@
+/** @jsx jsx */
+import { jsx } from 'theme-ui';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { ResizeHandlerData, ResizeHandlerProps } from './types';
+
+export const ResizeHandler: React.FC<ResizeHandlerProps> = props => {
+  const { dir, onStart, onEnd, onUpdate, className, style, children } = props;
+
+  const [isDragging, setIsDragging] = useState(false);
+  const listenersRef = useRef<ResizeHandlerData['listenersRef']>(null);
+
+  const handleMouseMove = useCallback(
+    e => {
+      onUpdate(e);
+    },
+    [onUpdate],
+  );
+
+  const cleanMouseListeners = useCallback(() => {
+    const oldRef = listenersRef.current;
+    if (oldRef) {
+      window.removeEventListener('mousemove', oldRef.handleMouseMove);
+      window.removeEventListener('touchmove', oldRef.handleMouseMove);
+      window.removeEventListener('mouseup', oldRef.handleMouseUp);
+      window.removeEventListener('touchend', oldRef.handleMouseUp);
+    }
+  }, []);
+
+  const handleMouseUp = useCallback(
+    e => {
+      setIsDragging(false);
+      cleanMouseListeners();
+      onEnd(e);
+    },
+    [cleanMouseListeners, onEnd],
+  );
+
+  const handleMouseDown = useCallback(
+    e => {
+      setIsDragging(true);
+      cleanMouseListeners();
+
+      listenersRef.current = {
+        handleMouseMove,
+        handleMouseUp,
+      };
+      window.addEventListener('mousemove', handleMouseMove);
+      window.addEventListener('touchmove', handleMouseMove);
+      window.addEventListener('mouseup', handleMouseUp);
+      window.addEventListener('touchend', handleMouseUp);
+
+      onStart(e);
+    },
+    [cleanMouseListeners, handleMouseMove, handleMouseUp, onStart],
+  );
+
+  useEffect(() => {
+    return () => {
+      cleanMouseListeners();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div
+      onMouseDown={handleMouseDown}
+      onTouchStart={handleMouseDown}
+      className={className}
+      style={{
+        cursor: `${dir}-resize`,
+        ...style,
```

Comment:
a) should we use `style` here or `_css` or `sm`?
b) would it make more sense to just spread rest props instead of a style prop?
c) should we be using a different component if we want to allow theme-ui props?

@jxnblk

Reply:
Generally, the `style` attribute is good to use for dynamic styles, but it doesn't look like that's really the case here, unless `dir` changes frequently. If `dir` is changing based on mouse events, etc., then `style` is fine to use here. For passing through `style` as props, it's sort of the same thing, but I haven't looked at where that's coming from. Does that make sense?

Reply:
yes! ok good, just wanted to make sure we weren't baking in anything that would "break" theme-ui implementations or user extensibility